api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/EriDSDCohortQueries.java

```diff
@@ -1283,103 +1283,48 @@ public CohortDefinition getBreastfeedingComposition() {
 
-    cd.addSearch(
-        "txCurr",
-        EptsReportUtils.map(
-            txCurrCohortQueries.getTxCurrCompositionCohort(cohortName, true),
-            "onOrBefore=${endDate},location=${location}"));
-    cd.addSearch(
-        "markedLastDispense",
-        EptsReportUtils.map(
-            getPatientsMarkedInLastDispenseQuery(),
-            "startDate=${startDate},endDate=${endDate},location=${location}"));
-    cd.setCompositionString("txCurr and markedLastDispense");
-    return cd;
-  }
+    CohortDefinition txCurr = txCurrCohortQueries.getTxCurrCompositionCohort(cohortName, true);
+    CohortDefinition communityDispense = getPatientsMarkedInLastCommunityDispense();
+    CohortDefinition nonPregnantNonBreastfeedingNotTB = getNonPregnantNonBreastfeedingAndNonTb();
 
-  /**
-   * N7 : Active ART patients marked in DC and are eligible
-   *
-   * @return @{@link CohortDefinition}
-   */
-  public CohortDefinition getActiveARTEligiblePatientsDC() {
-    CompositionCohortDefinition cd = new CompositionCohortDefinition();
-    cd.setName("N7 : Active patients in ART marked in last DC and are eligible");
-    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
-    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
-    cd.addParameter(new Parameter("location", "Location", Location.class));
+    String mappings = "onOrBefore=${endDate},location=${location}";
+    cd.addSearch("txCurr", EptsReportUtils.map(txCurr, mappings));
+
+    String dspMappings = "onOrAfter=${startDate},onOrBefore=${endDate},locationList=${location}";
+    cd.addSearch("communityDispense", EptsReportUtils.map(communityDispense, dspMappings));
 
     cd.addSearch(
-        "activeARTDC",
-        EptsReportUtils.map(
-            getActivePatientsOnARTDC(),
-            "startDate=${startDate},endDate=${endDate},location=${location}"));
-    cd.addSearch(
-        "d1EligeblePatients",
-        EptsReportUtils.map(
-            getAllPatientsWhoAreActiveAndStable(),
-            "startDate=${startDate},endDate=${endDate},location=${location}"));
-    cd.setCompositionString("activeARTDC AND d1EligeblePatients");
-    return cd;
-  }
+        "nonPregnantNonBreastfeedingNotTB", mapStraightThrough(nonPregnantNonBreastfeedingNotTB));
 
-  /**
-   * N7 : Active ART patients marked in DC and are unstable
-   *
-   * @return @{@link CohortDefinition}
-   */
-  public CohortDefinition getActiveInARTUnstableDC() {
-    CompositionCohortDefinition cd = new CompositionCohortDefinition();
-    cd.setName("N7 : Active in ART patients marked in last DC and are not eligible");
-    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
-    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
-    cd.addParameter(new Parameter("location", "Location", Location.class));
+    cd.setCompositionString("txCurr AND communityDispense AND nonPregnantNonBreastfeedingNotTB");
```

Comment:
@ynurmahomed can you please use PregnantAndBreastfeedingAndOnTBTreatment from develop
```suggestion
    cd.setCompositionString("txCurr AND communityDispense AND NOT PregnantAndBreastfeedingAndOnTBTreatment");
```